src/Symfony/Component/Filesystem/Watcher/INotifyWatcher.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Filesystem\Watcher;
+
+use Symfony\Component\Filesystem\Exception\IOException;
+
+/**
+ * @internal
+ */
+class INotifyWatcher implements WatcherInterface
+{
+    public function watch($path, callable $callback, float $timeout = null)
+    {
+        $inotifyInit = inotify_init();
+
+        if (false === $inotifyInit) {
+            throw new IOException('Unable initialize inotify', 0, null, $path);
+        }
+
+        stream_set_blocking($inotifyInit, false);
+
+        $isDir = is_dir($path);
+
+        if ($isDir) {
+            $watchId = inotify_add_watch($inotifyInit, $path, IN_CREATE | IN_DELETE | IN_MODIFY);
```

Comment:
The files moves (`IN_MOVE | IN_MOVE_SELF`) and attributes changes (`IN_ATTRIB`) should also be watched imo (and also `IN_DELETE_SELF`).